docs/reference/snapshot-restore/apis/get-snapshot-status-api.asciidoc

```diff
@@ -0,0 +1,266 @@
++
+Omitting a repository name retrieves a list of all snapshots currently running in the cluster.
+
+`<snapshot>`::
+(Required, string)
+Comma-separated list of snapshot names to retrieve status for.
+
+[role="child_attributes"]
+[[get-snapshot-status-api-request-body]]
+==== {api-request-body-title}
+
+`ignore_unavailable`::
+(Optional, boolean)
+If `false`, the request returns an error for any snapshots that are unavailable. Defaults to `false`.
++
+If `true`, the request ignores snapshots that are unavailable, such as those that are corrupted or temporarily cannot be returned.
+
+[role="child_attributes"]
+[[get-snapshot-status-api-response-body]]
+==== {api-response-body-title}
+
+`snapshot`::
+(string)
+Name of the snapshot.
+
+`uuid`::
+(string)
+Universally unique identifier (UUID) of the snapshot.
+
+`state`::
++
+--
+(string)
+Indicates the current snapshot state.
+
+.Values for `state`
+[%collapsible%open]
+====
+`FAILED`::
+  The snapshot finished with an error and failed to store any data.
+
+`IN_PROGRESS`::
+  The snapshot is currently running.
+
+`PARTIAL`::
+  The global cluster state was stored, but data of at least one shard was not stored successfully.
+  The <<get-snapshot-api-response-failures,`failures`>> section of the response contains more detailed information about shards
+  that were not processed correctly.
+
+`SUCCESS`::
+  The snapshot finished and all shards were stored successfully.
+====
+--
+
+`include_global_state`::
+(boolean)
+Indicates whether the current cluster state is included in the snapshot.
+
+[[get-snapshot-status-shards-stats]]
+`shards_stats`::
+(object)
+Contains a count of shards in the snapshot.
++
+.Properties of `shards_stats`
+[%collapsible%open]
+====
+`initializing`::
+(integer)
+Number of shards that are still initializing.
+
+`started`::
+(integer)
+Number of shards that have started but not are not finalized.
+
+`finalizing`::
+(integer)
+Number of shards that are finalizing but are not done.
+
+`done`::
+(integer)
+Number of shards that initialized, started, and finalized successfully.
+
+`failed`::
+(integer)
+Number of shards that failed to be included in the snapshot.
+
+`total`::
+(integer)
+Total number of shards included in the snapshot.
+====
+
+[[get-snapshot-status-stats]]
+`stats`::
+(object)
+Provides details on the number (`file_count`) and size (`size_in_bytes`) of files included in the snapshot.
++
+.Properties of `stats`
+[%collapsible%open]
+====
+`incremental`::
+(object)
+Number and size of files that are not already in the repository and need to be copied as part of the incremental snapshot.
+
+`processed`::
+(object)
+Number and size of files that are still being copied.
+
+`total`::
+(object)
+Total number and size of files that are referenced by the snapshot.
+
+`start_time_in_millis`::
+(long)
+The time, in milliseconds, when the snapshot creation process started.
+
+`time_in_millis`::
+(long)
+The total time, in milliseconds, that
```

Comment:
We generally try to avoid using Java-specific data types.

```suggestion
`start_time_in_millis`::
(integer)
The time, in milliseconds, when the snapshot creation process started.

`time_in_millis`::
(integer)
The total time, in milliseconds, that
```